src/components/ReviewForm.js

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from "react";
+import axios from "axios";
+
+export function ReviewForm(props) {
+  const { id } = props.editableReview;
+  const [errorTitleMessage, setErrorTitleMessage] = useState("");
+  const [errorScoreMessage, setErrorScoreMessage] = useState("");
+  const [errorBodyMessage, setErrorBodyMessage] = useState("");
+  const [errorReviewerMessage, setErrorReviewerMessage] = useState("");
+  const [formData, setFormData] = useState(
+    id
+      ? props.editableReview
+      : {
+          title: "",
+          score: 1,
+          body: "",
+          reviewer: "",
+        }
+  );
+  const actionText = id ? "レビューを編集する" : "レビューを登録する";
+
+  function handleChange({ target }) {
+    const value = target.value;
+    console.log(value);
+    setFormData({
+      ...formData,
+      [target.name]: value,
+    });
+  }
+
+  // async関数にしてあげる　updateReviewをawaitするために
+  async function handleSubmit(event) {
+    event.preventDefault(); // 画面遷移してしまわないように必要
+    console.log(formData);
+    if (validate(formData)) {
+      if (id) {
+        await updateReview(id);
+      } else {
+        postReview();
+      }
+    }
+    props.closeModal();
```

Comment:
validation 通らなかったらエラーメッセージ表示したいので closeModal しちゃうとまずそうです

```suggestion
    if (validate(formData)) {
      if (id) {
        await updateReview(id);
      } else {
        postReview();
      }
      props.closeModal();
    }
```

こうするか、もっと言えば guard clause にして

```suggestion
    if (!validate(formData)) {
      return;
    }

    if (id) {
      await updateReview(id);
    } else {
      postReview();
    }
    props.closeModal();
```

とするのがよさそうです